searxng/searxng
# Simple theme: add "Engine tokens" field.

## What does this PR do?

Add the missing ["Engine tokens"](https://searxng.github.io/searxng/admin/engines/private-engines.html) input field:

![image](https://user-images.githubusercontent.com/1594191/141468021-93c954fe-715b-4370-a401-b7c33a54c935.png)

![image](https://user-images.githubusercontent.com/1594191/141468083-e9c16116-8013-4ffe-9d39-fcb15d9f7776.png)

When selected:

![image](https://user-images.githubusercontent.com/1594191/141468147-620aefe6-40b0-406b-962e-dc138b4a01d0.png)

![image](https://user-images.githubusercontent.com/1594191/141468125-b6dc3315-76b5-4acc-9da2-ba7ab7561901.png)

## Why is this change important?





## How to test this PR locally?

* Check the tokens are saved
* Check the UI with different screen widths.

## Author's checklist

This part is hacky:
```css
#main_preferences .value select {
  width: 14rem;
}
#main_preferences .value input[type="text"] {
  width: 13.25rem;
```

I'm not sure how to fix that: `display: flex` on parent & `flex: 1` don't fix the width issue (some `select` element are too large)

## Related issues

* In the "User interface" tab, "Show advanced settings" is missing too. When enabled, the language / time selection / categories are shown in the front page (the user doesn't need to click a button to show these fields). The simple theme doesn't have that.
* In the "User interface" tab, "Style" is missing too (let the user choose the Oscar style).
